plextraktsync/trakt_api.py

```diff
@@ -169,12 +169,12 @@ def scrobbler(media: Union[Movie, TVEpisode]) -> ScrobblerProxy:
     @nocache
     @rate_limit()
     @time_limit()
-    def mark_watched(self, m, time):
+    def mark_watched(self, m, time, show_trakt_id=None):
         m.mark_as_seen(time)
         if m.media_type == "movies":
             self.watched_movies.add(m.trakt)
-        if m.media_type == "episodes":
-            self.watched_shows.add(TVShow(m.show).trakt, m.season, m.number)
+        if m.media_type == "episodes" and show_trakt_id:
```

Comment:
if show_trakt_id is not provided the episode is silently skipped. not sure if that may ever be an issue.